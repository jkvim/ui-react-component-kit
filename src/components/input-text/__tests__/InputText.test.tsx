import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { InputText } from '../';

describe('InputText', () => {

  describe('props', () => {

    const mockCallback = jest.fn();

    const wrapper = mount(
      <InputText
              id="id"
              label="label"
              placeholder="placeholder"
              value="value"
              hint="hint"
              errorMessage="error"
              showError={true}
              className="class"
              disabled={true}
              onBlur={mockCallback}
              onChange={mockCallback}
              onKeyDown={mockCallback}
      />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `id` in props', () => {
      expect(wrapper.props().id).toBe('id');
    });

    it('should pass `label` in props', () => {
      expect(wrapper.props().label).toBe('label');
    });

    it('should pass `placeholder` in props', () => {
      expect(wrapper.props().placeholder).toBe('placeholder');
    });

    it('should pass `value` in props', () => {
      expect(wrapper.props().value).toBe('value');
    });

    it('should pass `hint` in props', () => {
      expect(wrapper.props().hint).toBe('hint');
    });

    it('should pass `errorMessage` in props', () => {
      expect(wrapper.props().errorMessage).toBe('error');
    });

    it('should pass `showError` in props', () => {
      expect(wrapper.props().showError).toBe(true);
    });

    it('should pass `className` in props', () => {
      expect(wrapper.props().className).toBe('class');
    });

    it('should pass `disabled` in props', () => {
      expect(wrapper.props().disabled).toBe(true);
    });

    it('should pass `onBlur` in props', () => {
      expect(wrapper.props().onBlur).toBe(mockCallback);
    });

    it('should pass `onChange` in props', () => {
      expect(wrapper.props().onChange).toBe(mockCallback);
    });

    it('should pass `onKeyDown` in props', () => {
      expect(wrapper.props().onKeyDown).toBe(mockCallback);
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(<InputText id="id" label="label" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot error', () => {
    const wrapper = mount(<InputText id="id" label="label" errorMessage="Error" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot disabled', () => {
    const wrapper = mount(<InputText id="id" label="label" disabled={true} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('expect value to change when keys are pressed', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<InputText id="id" label="label" onChange={mockCallback} />);
    wrapper.find('input').at(0).simulate('change', { target: { value:  'a'}});
    // The mock function is called once
    expect(mockCallback).toBeCalled();
    // The new value was sent
    expect(mockCallback.mock.calls[0][0]).toBe('a');
  });

  it('expect everything to work when no onChange is provided', () => {
    const wrapper = mount(<InputText id="id" label="label" />);
    wrapper.find('input').at(0).simulate('change', { target: { value: 'a' } });
  });
});