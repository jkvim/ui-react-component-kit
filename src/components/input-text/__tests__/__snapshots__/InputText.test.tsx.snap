// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText should match snapshot 1`] = `
<InputText
  id="1234567890"
  label="The label"
>
  <styled.div>
    <div
      className="sc-iwsKbI hqgBrh"
    >
      <styled.label
        hasError={false}
        htmlFor="1234567890"
        theme={
          Object {
            "color": Object {
              "color": "#333333",
              "error": "#9f173f",
            },
          }
        }
      >
        <label
          className="sc-bdVaJa CyGsO"
          htmlFor="1234567890"
        >
          The label
        </label>
      </styled.label>
      <styled.input
        hasError={false}
        id="1234567890"
        innerRef={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        theme={
          Object {
            "color": Object {
              "active": "#333333",
              "border": "#333333",
              "color": "#333333",
              "disabled": "#acacac",
              "error": "#9f173f",
              "focus": "#333333",
            },
          }
        }
        type="text"
        value=""
      >
        <input
          className="sc-bxivhb kxesp"
          id="1234567890"
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </styled.input>
      <styled.div
        hasError={false}
        theme={
          Object {
            "color": Object {
              "error": "#9f173f",
            },
          }
        }
      >
        <div
          className="sc-jzJRlG jhccRS"
        />
      </styled.div>
    </div>
  </styled.div>
</InputText>
`;
